garrysmod/gamemodes/terrortown/gamemode/vgui/sb_team.lua

```diff
@@ -116,11 +101,41 @@ end
 
 function PANEL:UpdateSortCache()
    self.rows_sorted = {}
-   for k,v in pairs(self.rows) do
-      table.insert(self.rows_sorted, v)
+
+   for _, row in pairs(self.rows) do
+      table.insert(self.rows_sorted, row)
    end
 
-   table.sort(self.rows_sorted, CompareScore)
+   table.sort(self.rows_sorted, function(rowa, rowb)
+      local plya = rowa:GetPlayer()
+      local plyb = rowb:GetPlayer()
+
+      if not IsValid(plya) then return false end
+      if not IsValid(plyb) then return true end
+
+      local comp = 0 -- Lua doesnt have an Ordering enumeration, I think?
+      
+      local sort_mode = GetConVar("ttt_scoreboard_sorting"):GetString()
+      local sort_func = sboard_panel.sort_table[sort_mode]

```

Comment:
Ugh I am bad I should have investigated this more earlier.

I had uncorrectly assumed that because existing code in `sb_row.lua` and `sb_team.lua` used that variable without any nil checks, that it was something out of the scope of this PR. My bad.